traffic_ops/traffic_ops_golang/crconfig/handler.go

```diff
@@ -48,7 +49,8 @@ func Handler(w http.ResponseWriter, r *http.Request) {
 	defer inf.Close()
```

Comment:
I'm not understanding the purpose of this Handler, especially with the comment warning and the route definition:
`cdns/{cdn}/snapshot/new/?$`

How does this create a _new_ crconfig? Should this route definition continue to be a part of the clients going forward as well? I see that clients have both available APIs to call from (`getCRConfig` and `GetCRConfigNew`). Outside of tests, I don't see the `GetCRConfigNew` being utilized (granted we have no way of know what a client is actually utilizing.

Reply:
This constructs - but does not store - the CDN Snapshot, and returns it to the client. Other components "MUST NOT" use it because it represents what the new CDN Snapshot *would be* **if** one were taken. Components MUST use the stored Snapshot, as that is the configuration that is intended to be active.

`GetCRConfigNew` may not be used directly, but the Go client is not the only supported client of the TO API; TP uses it to display the changes to the CRConfig (ideally it would show the entire Snapshot, but that's another matter entirely) that would be made if a new Snapshot were taken, which is absolutely vital to operators.

Reply:
Ahh ok.